pkg/intercp/catalog/heartbeat_component.go

```diff
@@ -110,7 +114,7 @@ func (h *heartbeatComponent) heartbeat(ctx context.Context, ready bool) bool {
 	}
 	resp, err := client.Ping(ctx, h.request)
 	if err != nil {
-		heartbeatLog.Error(err, "could not send a heartbeat to a leader")
+		heartbeatLog.Info("could not send a heartbeat to a leader. It's likely due to leader change", "cause", err)
```

Comment:
Info is fine, because we then do `h.leader` so on the next iteration we will try to reconnect to leader. If problem persists and it's not caused by no leader then we will log error in this line
```
heartbeatLog.Error(err, "could not connect to leader")
```